src/ucp/core/ucp_ep.c

```diff
@@ -259,6 +259,14 @@ static void ucp_ep_destroy_base(ucp_ep_h ep)
     ucs_assert(ep->discard_refcount == 0);
 
     ucp_ep_remove_progress_callbacks(ep);
+
+    /* Release UCP EP ID after all outstanding/pending operations are purged
+     * (e.g. using discarding) or flushed (i.e. UCP EP refcount reaches '0')
+     * and all progress callbacks are descheduled to avoid trying to get UCP EP
+     * ID for some operation (e.g. getting UCP EP ID for sending WIREUP_MSG/ACK
+     * from a progress callback) */
+    ucp_ep_set_failed(ep);
```

Comment:
why need to set UCP_EP_FLAG_FAILED flag here?

Reply:
- move this to non-internal-ep destroy function
- remove UCP_EP_FLAG_INTERNAL check

Reply:
unfortunately, TMP EPs are deleted by the same function as normal UCP EPs, i.e. `ucp_ep_disconnected()`